src/modules/musig/tests_impl.h

```diff
@@ -950,6 +961,191 @@ void musig_tweak_test(secp256k1_scratch_space *scratch) {
+        CHECK(secp256k1_memcmp_var(&pk_test[i], &pk[i], sizeof(pk_test[i])) == 0);
+    }
+}
+
+void permute(size_t *arr, size_t n) {
+    size_t i;
+    for (i = n - 1; i >= 1; i--) {
+        size_t tmp, j;
+        j = secp256k1_testrand_int(i + 1);
+        tmp = arr[i];
+        arr[i] = arr[j];
+        arr[j] = tmp;
+    }
+}
+
+void rand_xonly_pk(secp256k1_xonly_pubkey *pk) {
+    unsigned char seckey[32];
+    secp256k1_testrand256(seckey);
+    CHECK(secp256k1_xonly_pubkey_create(pk, seckey));
+}
+
+void musig_sort_test(void) {
+    secp256k1_xonly_pubkey pk[5];
+    unsigned char pk_ser[5][32];
+    int i;
+    size_t pk_order[5] = { 0, 1, 2, 3, 4 };
+
+    for (i = 0; i < 5; i++) {
+        memset(pk_ser[i], 0, sizeof(pk_ser[i]));
+    }
+    pk_ser[0][0] = 5;
+    pk_ser[1][0] = 8;
+    pk_ser[2][0] = 0x0a;
+    pk_ser[3][0] = 0x0b;
+    pk_ser[4][0] = 0x0c;
+    for (i = 0; i < 5; i++) {
+        CHECK(secp256k1_xonly_pubkey_parse(ctx, &pk[i], pk_ser[i]));
+    }
+
+    permute(pk_order, 1);
+    musig_sort_test_helper(pk, pk_order, 1);
+    permute(pk_order, 2);
+    musig_sort_test_helper(pk, pk_order, 2);
+    permute(pk_order, 3);
+    musig_sort_test_helper(pk, pk_order, 3);
+    for (i = 0; i < count; i++) {
+        permute(pk_order, 4);
+        musig_sort_test_helper(pk, pk_order, 4);
+    }
+    for (i = 0; i < count; i++) {
+        permute(pk_order, 5);
+        musig_sort_test_helper(pk, pk_order, 5);
+    }
+    /* Check that sorting also works for random pubkeys */
+    for (i = 0; i < count; i++) {
+        int j;
+        for (j = 0; j < 5; j++) {
+            rand_xonly_pk(&pk[j]);
+        }
+        secp256k1_musig_sort_pubkeys(ctx, pk, 5);
+        for (j = 1; j < 5; j++) {
+            CHECK(secp256k1_memcmp_var(&pk[j - 1], &pk[j], sizeof(pk[j])) <= 0);
+        }
+    }
+}
+
+void musig_test_vectors_helper(unsigned char pk_ser[][32], int n_pks, const unsigned char *combined_pk_expected, int sort, int has_second_pk, int second_pk_idx) {
+    secp256k1_xonly_pubkey *pk = malloc(n_pks * sizeof(secp256k1_xonly_pubkey));
+    secp256k1_xonly_pubkey combined_pk;
+    unsigned char combined_pk_ser[32];
+    secp256k1_musig_pre_session pre_session;
+    secp256k1_fe second_pk_x;
+    int i;
+
+    for (i = 0; i < n_pks; i++) {
+        CHECK(secp256k1_xonly_pubkey_parse(ctx, &pk[i], pk_ser[i]));
+    }
+
+    if (sort) {
+        CHECK(secp256k1_musig_sort_pubkeys(ctx, pk, n_pks));
+    }
+    CHECK(secp256k1_musig_pubkey_combine(ctx, NULL, &combined_pk, &pre_session, pk, n_pks) == 1);
+    CHECK(secp256k1_fe_set_b32(&second_pk_x, pre_session.second_pk));
+    CHECK(secp256k1_fe_is_zero(&second_pk_x) == !has_second_pk);
+    if (!secp256k1_fe_is_zero(&second_pk_x)) {
+        CHECK(secp256k1_memcmp_var(&pk_ser[second_pk_idx], &pre_session.second_pk, sizeof(pk_ser[second_pk_idx])) == 0);
+    }
+    CHECK(secp256k1_xonly_pubkey_serialize(ctx, combined_pk_ser, &combined_pk));
+    /* TODO: remove */
```

Comment:
what's the point of keeping this commented code around?

Reply:
The point is that it's easy to print test vectors if we change them. I updated the comment to make that more clear.